src/main/java/org/breedinginsight/brapps/importer/services/processors/GermplasmProcessor.java

```diff
@@ -242,52 +259,88 @@ public Map<String, ImportPreviewStatistics> process(List<BrAPIImport> importRows
+            if (germplasm.getEntryNo() == null) {
+                germplasm.setEntryNo(Integer.toString(i + 1));
+            } else {
+                userProvidedEntryNumbers.add(germplasm.getEntryNo());
+            }
+            entryNumberCounts.put(germplasm.getEntryNo(),
+                    entryNumberCounts.containsKey(germplasm.getEntryNo()) ? entryNumberCounts.get(germplasm.getEntryNo()) + 1 : 1);
+
+            UUID importListId = brAPIGermplasmService.getGermplasmListId(importList);
+
             // Germplasm
-            if (germplasm != null && germplasm.getGermplasmName() != null) {
 
-                // Get the breeding method database object
-                ProgramBreedingMethodEntity breedingMethod = null;
-                if (germplasm.getBreedingMethod() != null) {
-                    if (breedingMethods.containsKey(germplasm.getBreedingMethod())) {
-                        breedingMethod = breedingMethods.get(germplasm.getBreedingMethod());
+            //todo double check what dbgermplasmbyaccessionNo actually getting
+            //TODO maybe make separate method for cleanliness
+            if (germplasm != null) {
+                //Fetch and update existing germplasm
+                BrAPIGermplasm existingGermplasm;
+
+                // Have GID so updating an existing germplasm record
+                if (germplasm.getAccessionNumber() != null) {
+                    if (dbGermplasmByAccessionNo.containsKey(germplasm.getAccessionNumber()) ) {
+                        existingGermplasm = dbGermplasmByAccessionNo.get(germplasm.getAccessionNumber());
                     } else {
-                        List<ProgramBreedingMethodEntity> breedingMethodResults = breedingMethodDAO.findByNameOrAbbreviation(germplasm.getBreedingMethod(), program.getId());
-                        if (breedingMethodResults.size() > 0) {
-                            breedingMethods.put(germplasm.getBreedingMethod(), breedingMethodResults.get(0));
+                        ValidationError ve = new ValidationError("GID", missingGID, HttpStatus.NOT_FOUND);
+                        validationErrors.addError(i+2, ve );  // +2 instead of +1 to account for the column header row.
+                        continue;
+                    }
+
+                    // Has an existing pedigree so add error and skip
+                    // no pedigree on second import, don't change existing
+                    // if there is pedigree & previously not, upgrade pedigree
+                    // if previous pedigree & different pedigree, validation error
+                    if (!StringUtils.isBlank(existingGermplasm.getPedigree())) {
+                        ValidationError ve = new ValidationError("Pedigree", pedigreeAlreadyExists, HttpStatus.UNPROCESSABLE_ENTITY);
+                        validationErrors.addError(i+2, ve );  // +2 instead of +1 to account for the column header row.
+                        continue;
+                    }
```

Comment:
If a germplasm record has pedigree, and I include it in a second upload to append synonyms, it looks like this check would prevent that from happening.  Is that correct?

Reply:
Ya thanks, I updated the logic. 

Error conditions:
- has existing pedigree and file pedigree is different

Valid conditions:
- no existing pedigree and file different pedigree (not blank though, will fail other validations)
- existing pedigree and file pedigree same